pkg/inventory/policy.go

```diff
@@ -119,7 +122,7 @@ func CanPrune(inv InventoryInfo, obj *unstructured.Unstructured, policy Inventor
 	matchStatus := inventoryIDMatch(inv, obj)
 	switch matchStatus {
 	case Empty:
-		return false
+		return true
```

Comment:
So should we maybe use the inventory policy here to determine what we should do?

Reply:
We can use the inventory policy here. Among the three policies, the `AdoptAll` and `AdoptIfNoInventory` allows applying in this case, so we can also allow pruning for them. 

The motivation of this PR is that we need to think about how to handle the first apply after the cli is switched to the new library version with inventory policy. If the cli runs `AdoptIfNoInventory` for the first apply, then all resources can be either applied with owning-inventory annotation or be pruned. The following apply with `InventoryPolicyMustMatch` would work.

Reply:
Updated this in the 2nd commit.